### DISCUSSION TITLE
summarize doesn't show everything

### DISCUSSION THREAD

**Discussion by mikepictor** [TimefoldAI/timefold-solver#1159] [General]
I am curious why `solutionManager.analyze(solution).summarize()` doesn't show everything. I can see in the source
```
matches.stream().sorted(matchScoreComparator).limit(3L).forEach((match) -> {
         summary.append("%12s%s: justified with (%s)\n".formatted(" ", match.score().toShortString(), match.justification()));
});
if (matches.size() > 3) {
         summary.append("%12s%s\n".formatted(" ", "..."));
}
```
Why the 3 limit? I have had to code up my own summary function just to get all the matches to show in my log.

**Comment by triceo** [comment]
In my opinion, there is a limit to how many matches you can see for the output to still be useful. Having even a couple dozen matches in a `toString()` wouldn't be helpful. You can't display this anywhere, and arguably you can't even make sense of it - above a certain number, it is only useful to computers, not humans.

Is 3 the right limit? Maybe not, but there clearly is a numeric limit and the specific number will depend on individual preference and circumstance. The score analysis interface does give you all the raw information, so - as you point out yourself - you can write any sort of visualization you prefer.
